confluent_kafka/avro/serializer/__init__.py

```diff
@@ -33,8 +61,270 @@ def __str__(self):
 
 
 class KeySerializerError(SerializerError):
-    pass
+    def __new__(cls, message):
+        return super(KeySerializerError, cls).__new__(SerializerError, message, True)
 
 
 class ValueSerializerError(SerializerError):
     pass
+
+
+class ContextStringIO(io.BytesIO):
+    """
+    Wrapper to allow use of StringIO via 'with' constructs.
+    """
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        self.close()
+        return False
+
+
+def TopicNameStrategy(topic=None, is_key=False, schema=None):
```

Comment:
`DefaultSubjectNameStrategy` is more descriptive.

Reply:
oh, i see you have others. `TopicNameStrategy` makes sense.

Reply:
All naming conventions come directly from the Java implementation to keep things simple.